PR: https://github.com/odoo/odoo/pull/186906

From: 9144eb051538d5a92d403e1ad9c635ce39e090ec
From: Gorash
Date: 2024-11-14 10:39:02

Breaking data model changes scores: del:9 + add:0, change matches:
-    _inherit = ['hr.employee.public']
-    _inherit = ["hr.employee"]
-    _inherit = ['ir.ui.menu']
-    _inherit = ["res.company"]
-    _inherit = ['res.config.settings']
-    _inherit = ['res.users']
-    _inherit = ['resource.calendar']
-    _inherit = ['resource.calendar.leaves']
-    _inherit = ['resource.resource']

Total Changes: 22

[REV] *: Restore the model `_name`

Following internal discussions, the python typing and inheritance task is
canceled. The use of the class name instead of '_name' was done to allow
notations such as `partner_id = fields.Many2one[base.ResParter](string='Partner')`.
Since this notation is canceled, we put `_name` back (although the class
name is sufficient).

Partial revert:
https://github.com/odoo/enterprise/pull/69762
https://github.com/odoo/odoo/pull/178200

closes odoo/odoo#186906

Related: odoo/enterprise#73643
Signed-off-by: Raphael Collet <rco@odoo.com>

================================= pseudo patch: =================================

--- a/addons/hr_contract/models/hr_contract.py
+++ b/addons/hr_contract/models/hr_contract.py
@@ -16,6 +16,7 @@ _logger = logging.getLogger(__name__)
 
 
 class HrContract(models.Model):
+    _name = 'hr.contract'
     _description = 'Contract'
     _inherit = ['mail.thread', 'mail.activity.mixin']
     _mail_post_access = 'read'

--- a/addons/hr_contract/models/hr_employee.py
+++ b/addons/hr_contract/models/hr_employee.py
@@ -13,7 +13,7 @@ from odoo.exceptions import UserError
 
 
 class HrEmployeePublic(models.Model):
-    _inherit = ['hr.employee.public']
+    _inherit = 'hr.employee.public'
 
     first_contract_date = fields.Date(compute='_compute_manager_only_fields', search='_search_first_contract_date')
 
@@ -26,7 +26,7 @@ class HrEmployeePublic(models.Model):
 
 
 class HrEmployeeBase(models.AbstractModel):
-    _inherit = ["hr.employee.base"]
+    _inherit = "hr.employee.base"
 
     @api.model
     def _get_new_hire_field(self):
@@ -34,7 +34,7 @@ class HrEmployeeBase(models.AbstractModel):
 
 
 class HrEmployee(models.Model):
-    _inherit = ["hr.employee"]
+    _inherit = "hr.employee"
 
     legal_name = fields.Char(compute='_compute_legal_name', store=True, readonly=False, groups="hr.group_hr_user")
     vehicle = fields.Char(string='Company Vehicle', groups="hr.group_hr_user")

--- a/addons/hr_contract/models/hr_payroll_structure_type.py
+++ b/addons/hr_contract/models/hr_payroll_structure_type.py
@@ -5,6 +5,7 @@ from odoo import fields, models
 
 
 class HrPayrollStructureType(models.Model):
+    _name = 'hr.payroll.structure.type'
     _description = 'Salary Structure Type'
 
     name = fields.Char('Salary Structure Type')

--- a/addons/hr_contract/models/ir_ui_menu.py
+++ b/addons/hr_contract/models/ir_ui_menu.py
@@ -5,7 +5,7 @@ from odoo import models
 
 
 class IrUiMenu(models.Model):
-    _inherit = ['ir.ui.menu']
+    _inherit = 'ir.ui.menu'
 
     def _load_menus_blacklist(self):
         res = super()._load_menus_blacklist()

--- a/addons/hr_contract/models/res_company.py
+++ b/addons/hr_contract/models/res_company.py
@@ -5,7 +5,7 @@ from odoo import fields, models
 
 
 class ResCompany(models.Model):
-    _inherit = ["res.company"]
+    _inherit = "res.company"
 
     contract_expiration_notice_period = fields.Integer("Contract Expiry Notice Period", default=7)
     work_permit_expiration_notice_period = fields.Integer("Work Permit Expiry Notice Period", default=60)

--- a/addons/hr_contract/models/res_config_settings.py
+++ b/addons/hr_contract/models/res_config_settings.py
@@ -5,7 +5,7 @@ from odoo import fields, models
 
 
 class ResConfigSettings(models.TransientModel):
-    _inherit = ['res.config.settings']
+    _inherit = 'res.config.settings'
 
     contract_expiration_notice_period = fields.Integer(string="Contract Expiry Notice Period", related='company_id.contract_expiration_notice_period', readonly=False)
     work_permit_expiration_notice_period = fields.Integer(string="Work Permit Expiry Notice Period", related='company_id.work_permit_expiration_notice_period', readonly=False)

--- a/addons/hr_contract/models/res_users.py
+++ b/addons/hr_contract/models/res_users.py
@@ -5,7 +5,7 @@ from odoo import models, fields, api, _
 
 
 class ResUsers(models.Model):
-    _inherit = ['res.users']
+    _inherit = 'res.users'
 
     vehicle = fields.Char(related="employee_id.vehicle")
     bank_account_id = fields.Many2one(related="employee_id.bank_account_id")

--- a/addons/hr_contract/models/resource.py
+++ b/addons/hr_contract/models/resource.py
@@ -7,7 +7,7 @@ from odoo.osv.expression import AND
 
 
 class ResourceCalendar(models.Model):
-    _inherit = ['resource.calendar']
+    _inherit = 'resource.calendar'
 
     contracts_count = fields.Integer("# Contracts using it", compute='_compute_contracts_count', groups="hr_contract.group_hr_contract_manager")
 

--- a/addons/hr_contract/models/resource_calendar_leaves.py
+++ b/addons/hr_contract/models/resource_calendar_leaves.py
@@ -7,7 +7,7 @@ from odoo import models
 
 
 class ResourceCalendarLeaves(models.Model):
-    _inherit = ['resource.calendar.leaves']
+    _inherit = 'resource.calendar.leaves'
 
     def _compute_calendar_id(self):
         def date2datetime(date, tz):

--- a/addons/hr_contract/models/resource_resource.py
+++ b/addons/hr_contract/models/resource_resource.py
@@ -9,7 +9,7 @@ from odoo.addons.resource.models.utils import Intervals
 
 
 class ResourceResource(models.Model):
-    _inherit = ['resource.resource']
+    _inherit = 'resource.resource'
 
     def _get_calendars_validity_within_period(self, start, end, default_company=None):
         assert start.tzinfo and end.tzinfo
